int32 tokens
bool sim
---
int8[] state
